src/frontend/src/handler/create_source.rs

```diff
@@ -119,8 +128,11 @@ fn json_schema_infer_use_schema_registry(schema_config: &Option<(AstString, bool
 async fn extract_avro_table_schema(
     info: &StreamSourceInfo,
     with_properties: &HashMap<String, String>,
+    row_options: &mut BTreeMap<String, String>,
```

Comment:
We use &mut for `row_options` because in the `bind_columns_from_source` function fields of options are consumed (i.e. row_options.remove(&key)). And we will check `options.is_empty()` in the end.
https://github.com/risingwavelabs/risingwave/blob/main/src/frontend/src/handler/create_source.rs#L572-L584

Reply:
> The difference between the two's type might be confusing (especially mut). 

We hope to consume both configs here but it seems impossible for connector props. Let's keep `format_encode_options` safe from unknown fields.